src/components/event_view/event_fields/EventFieldEdit.js

```diff
@@ -0,0 +1,180 @@
+import PropTypes from 'prop-types';
+import Form from 'react-bootstrap/Form';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import CurrencyInput from 'react-currency-input-field';
+import './EventFieldEdit.css';
+import { FormInput } from '../../global/FormInput';
+import { formatDateForUser } from '../../../shared/dateFormatter';
+import { usdFormatter } from '../../../shared/currencyFormatter';
+
+/**
+ * EventFieldEdit is a visual element for an event, which receives and displays
+ * a field type and value.
+ */
+
+export default class EventFieldEdit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.field.value
+    };
+
+    this.handleTextChange = this.handleTextChange.bind(this);
+    this.handleCostChange = this.handleCostChange.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+  }
+
+  /***************************** Core functions *****************************/
+  /**
+   * Handles the change of an events date.
+   * @param e the event containing the new text.
+   */
+  handleTextChange(e) {
+    this.setState({ value: e.target.value });
+  }
+
+  /**
+   * Handles the change of an events date.
+   * @param date the new date.
+   */
+  handleDateChange(date) {
+    this.setState({ value: date.toString() });
+  }
+
+  /**
+   * Handles the change of an events cost.
+   * @param cost the new cost.
+   */
+  handleCostChange(cost) {
+    this.setState({ value: usdFormatter.format(cost) });
+  }
+
+  /**************************** Visual component ****************************/
+  render() {
+    let inputField;
+
+    switch (this.props.field.name) {
+      case 'cost':
+        inputField = this.renderCurrencyInput();
+        break;
+      case 'begin_time':
+        inputField = this.renderRestrictedDateSelection();
+        break;
+      case 'end_time':
+        inputField = this.renderUnrestrictedDateSelection();
+        break;
+      default:
+        inputField = this.renderTextInput();
+    }
+
+    return (
+      <td>
+        {inputField}
+        <button
+          className={'btn btn-primary editSave'}
+          onClick={() => this.props.onClick(this.props.field, this.state.value)}
+        >
+          <b>Save changes</b>
+        </button>
+        <button
+          className={'btn btn-secondary editCancel'}
+          onClick={() => this.props.onClick(this.props.field)}
+        >
+          <b>Cancel</b>
+        </button>
+      </td>
+    );
+  }
+
+  /**
+   * Renders a text input field.
+   * @returns {*} the text input field.
+   */
+  renderTextInput() {
+    return (
+      <Form className={'textInput'}>
+        <FormInput
+          displayName={this.props.field.type}
+          name={this.props.field.name}
+          type={this.props.field.type}
+          handleChange={this.handleTextChange}
+          value={this.state.value}
+        />
+      </Form>
+    );
+  }
```

Comment:
I spent some time thinking about this and didn't come to a definite conclusion. I could break this file down into EventFieldEdit and some mini-components, but I'm not sure if we'll be reusing any of those mini-components again. So, should they be given their own files? Or should they just be considered as subcomponents of this specific component? I'm open to either option.

Reply:
I feel like given that they each need their own render, they should be their own components... Even if they aren't reused this feels like a single responsibility violation

Reply:
I think we should discuss this as a group because I'm not sold on this idea. I understand the logic, but the idea of creating additional files for render methods that only serve the purpose of servicing one component seems unnecessary.

Reply:
From what I've read, both approaches are legitimate. When React was introduced, they said it was up to developers to decide where to divide the different components. One realization I had when reading this code was that creating different components may not actually shorten the code as much as I thought, because there are so many properties we'd have to pass in to the mini render components. I think both approaches are fine, but yes, some discussion wouldn't be bad!